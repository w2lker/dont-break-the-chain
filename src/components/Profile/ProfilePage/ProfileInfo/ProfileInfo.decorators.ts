import withStyles from '@material-ui/core/styles/withStyles';
import { connect } from 'react-redux';

import ProfileInfo, { IProfileInfoProps } from './ProfileInfo';
import profileInfoStyles from './ProfileInfo.styles';

import { IProfileReducer } from '../../../../models/profile';

const mapStateToProps = (state: any): IProfileInfoProps => {
  const profile: IProfileReducer = state.profile;
  return {
    avatar: profile.avatar,
    name: profile.name,
    email: profile.email,
  };
};

const mapDispatchToProps = null;

const connectedComponent = connect(mapStateToProps, mapDispatchToProps)(ProfileInfo);

const styledComponent = withStyles(profileInfoStyles)(connectedComponent);

export default styledComponent;
