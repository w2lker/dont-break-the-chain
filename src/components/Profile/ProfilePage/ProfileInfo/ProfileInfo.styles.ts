import { bon } from '../../../../utils/styleUtils';

const profileInfoStyles = {
  wrapper: {
    margin: '30px 30px 0',
    padding: '16px',
    boxShadow: '0px 4px 8px rgba(176, 190, 197, 0.24)',
  },
  avatar: {
    ...bon.size(48),
    backgroundPosition: 'center center',
    backgroundSize: 'cover',
    borderRadius: '50%',
  },
  name: {
    ...bon.setFont(20, '#263228', 500, '0.15px'),
    margin: '12px 0 0',
    lineHeight: '24px',
  },
  email: {
    ...bon.setFont(14, 'rgba(38, 50, 56, 0.72)', 500, '0.15px'),
    lineHeight: '20px',
    margin: 0,
  },
};

export default profileInfoStyles;
