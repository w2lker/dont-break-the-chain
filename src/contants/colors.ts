import colorBlue from '@material-ui/core/colors/blue';
import colorBrown from '@material-ui/core/colors/brown';
import colorCyan from '@material-ui/core/colors/cyan';
import colorDeepPurple from '@material-ui/core/colors/deepPurple';
import colorGreen from '@material-ui/core/colors/green';
import colorIndigo from '@material-ui/core/colors/indigo';
import colorOrange from '@material-ui/core/colors/orange';
import colorPink from '@material-ui/core/colors/pink';
import colorPurple from '@material-ui/core/colors/purple';
import colorTeal from '@material-ui/core/colors/teal';

export type colorsKeys = 'pink' | 'purple' | 'deepPurple' | 'indigo' | 'blue' | 'cyan' | 'teal' | 'green' | 'orange' | 'brown';

export const colors: {[color: string]: any}  = {
  blue: colorBlue,
  brown: colorBrown,
  pink: colorPink,
  cyan: colorCyan,
  deepPurple: colorDeepPurple,
  indigo: colorIndigo,
  green: colorGreen,
  orange: colorOrange,
  purple: colorPurple,
  teal: colorTeal,
};
