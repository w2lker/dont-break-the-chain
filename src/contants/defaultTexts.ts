const defaultTexts = {
  navigationMenu: {
    chains: 'Chains',
    habits: 'Habits',
    profile: 'Profile',
  },
  profile: {
    header: {
      title: 'Profile',
    },
  },
};

export default defaultTexts;
