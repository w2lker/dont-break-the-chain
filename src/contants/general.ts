import { RequestStatuses } from '../models/general';

// eslint-disable-next-line
export const requestStatuses = {
  active: 'active',
  success: 'success',
  error: 'error',
};
