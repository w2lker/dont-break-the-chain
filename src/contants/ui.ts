export const defaultIcons = {
  navigationMenu: {
    chains: 'layers',
    habits: 'flag-checkered',
    profile: 'account',
  },
};

// TODO: provide real placeholder image and attach link to it here
export const avatarPlaceholder = './placeholder.png';
