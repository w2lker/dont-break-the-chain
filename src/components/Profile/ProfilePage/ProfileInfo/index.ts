import ProfileInfo from './ProfileInfo.decorators';

export default ProfileInfo;
