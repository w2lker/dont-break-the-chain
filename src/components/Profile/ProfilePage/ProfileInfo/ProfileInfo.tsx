import * as React from 'react';
import { avatarPlaceholder } from '../../../../contants/ui';

export interface IProfileInfoProps {
  avatar?: string;
  name?: string;
  email?: string;
  classes?: any;
}

const ProfileInfo: React.FC<IProfileInfoProps> = (props) => {
  const { name, email, classes } = props;
  const avatar = props.avatar || avatarPlaceholder;
  const avatarStyles = {
    backgroundImage: `url(${avatar})`,
  };

  return (
    <div className={classes.wrapper}>
      <div className={classes.avatar} style={avatarStyles} />
      <h2 className={classes.name}>{name}</h2>
      <p className={classes.email}>{email}</p>
    </div>
  );
};

ProfileInfo.defaultProps = {
  name: '',
  email: '',
  classes: {
    wrapper: '',
    avatar: '',
    name: '',
    email: '',
  },
};

export default ProfileInfo;
