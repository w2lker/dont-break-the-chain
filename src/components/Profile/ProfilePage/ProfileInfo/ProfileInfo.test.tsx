import * as React from 'react';

import { mount, shallow } from 'enzyme';

import { avatarPlaceholder } from '../../../../contants/ui';
import ProfileInfo, { IProfileInfoProps } from './ProfileInfo';
import profileInfoStyles from './ProfileInfo.styles';

const classes = {
  wrapper: 'profile-info-wrapper',
  avatar: 'profile-info-avatar',
  name: 'profile-info-name',
  email: 'profile-info-email',
};

const classesKeys = Object.keys(classes);

describe('ProfileInfo component', () => {
  const emptyProps: IProfileInfoProps = {
    classes,
  };

  const sampleProps: IProfileInfoProps = {
    classes,
    avatar: './some-avatar.png',
    name: 'Sandra Adams',
    email: 'purchasenow@gmail.com',
  };

  it('match snapshot', () => {
    const component = mount(<ProfileInfo {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  describe('renders with empty props', () => {
    const component = shallow(<ProfileInfo {...emptyProps} />);
    it('renders wrapper', () => {
      const wrapper = component.find(`.${classes.wrapper}`);
      expect(wrapper.length).toBe(1);
    });

    it('renders avatar with placeholder', () => {
      const avatar = component.find(`.${classes.avatar}`);
      expect(avatar.length).toBe(1);
      const a = 3;
      // @ts-ignore
      expect(avatar.props().style.backgroundImage).toEqual(expect.stringContaining(avatarPlaceholder));
    });

    it('renders with empty name', () => {
      const name = component.find(`.${classes.name}`);
      expect(name.length).toBe(1);
      expect(name.text()).toBe('');
    });

    it('renders with empty email', () => {
      const email = component.find(`.${classes.email}`);
      expect(email.length).toBe(1);
      expect(email.text()).toBe('');
    });
  });

  describe('renders with sample props', () => {
    const component = shallow(<ProfileInfo {...sampleProps} />);
    it('renders wrapper', () => {
      const wrapper = component.find(`.${classes.wrapper}`);
      expect(wrapper.length).toBe(1);
    });
    it('renders avatar', () => {
      const avatar = component.find(`.${classes.avatar}`);
      expect(avatar.length).toBe(1);
      // @ts-ignore
      expect(avatar.props().style.backgroundImage).toEqual(expect.stringContaining(sampleProps.avatar));
    });

    it('renders name', () => {
      const name = component.find(`.${classes.name}`);
      expect(name.length).toBe(1);
      expect(name.text()).toBe(sampleProps.name);
    });

    it('renders with email', () => {
      const email = component.find(`.${classes.email}`);
      expect(email.length).toBe(1);
      expect(email.text()).toBe(sampleProps.email);
    });
  });
});

describe('ProfileInfo styles', () => {
  it('should contain principle classes', () => {
    expect(profileInfoStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(profileInfoStyles[keyValue]).toBeDefined();
    });
  });
});

describe('ProfileInfo decorators', () => {
  test.todo('Check classes to be passed to the child component');
  test.todo('Check data to be passed from redux store (avatar, name, mail)');
});
