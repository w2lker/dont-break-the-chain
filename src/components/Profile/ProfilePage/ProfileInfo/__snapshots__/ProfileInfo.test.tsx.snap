// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo component match snapshot 1`] = `
"<ProfileInfo classes={{...}} avatar=\\"./some-avatar.png\\" name=\\"Sandra Adams\\" email=\\"purchasenow@gmail.com\\">
  <div className=\\"profile-info-wrapper\\">
    <div className=\\"profile-info-avatar\\" style={{...}} />
    <h2 className=\\"profile-info-name\\">
      Sandra Adams
    </h2>
    <p className=\\"profile-info-email\\">
      purchasenow@gmail.com
    </p>
  </div>
</ProfileInfo>"
`;
